jgrapht-core/src/main/java/org/jgrapht/intervalgraph/CenteredIntervalTree.java

```diff
@@ -0,0 +1,110 @@
+package org.jgrapht.intervalgraph;
+
+import org.jgrapht.Graph;
+import org.jgrapht.intervalgraph.interval.Interval;
+import org.omg.SendingContext.RunTime;
+
+import java.util.*;
+
+public class CenteredIntervalTree<T extends Comparable<T>> implements IntervalGraph<T>{
+    private T centerPoint;
+    private CenteredIntervalTree<T> leftTree; // contains all the intervals *completely* to the left of the center point
+    private CenteredIntervalTree<T> rightTree; // contains all the intervals *completely* to the right of the center point
+    private List<Interval<T>> intersectionsByStart = new LinkedList<>(); // intervals intersecting center point, sorted by start point
+    private List<Interval<T>> intersectionsByEnd = new LinkedList<>(); // ..., sorted by end point
+
+    public CenteredIntervalTree(T centerPoint, List<Interval<T>> intervals) {
+        this.centerPoint = centerPoint;
+        compute(intervals);
+    }
+
+    public CenteredIntervalTree(List<Interval<T>> intervals) {
+        this.centerPoint = intervals.get(0).getEnd(); // TODO replace this with something, its arbitrary
+        compute(intervals);
+    }
+
+    private void compute( List<Interval<T>> intervals) {
+        List<Interval<T>> leftIntervals = new LinkedList<>(); // containing the intervals completely to the left of the center point
+        List<Interval<T>> rightIntervals = new LinkedList<>(); // ... to the right
+        List<Interval<T>> intersectingIntervals = new LinkedList<>(); // intervals intersecting the center point
+
+        for (Interval<T> interval: intervals) {
+            if (interval.contains(centerPoint)) {
+                intersectingIntervals.add(interval);
```

Comment:
Perhaps we can again benefit from some existing libraries to compute list intersection. https://commons.apache.org/proper/commons-collections/javadocs/api-2.1.1/org/apache/commons/collections/ListUtils.html